spring-cloud-gcp-pubsub/src/main/java/org/springframework/cloud/gcp/pubsub/core/subscriber/PubSubSubscriberTemplate.java

```diff
@@ -171,10 +195,23 @@ private void ack(String subscriptionName, Collection<String> ackIds) {
 		this.subscriberStub.acknowledgeCallable().call(acknowledgeRequest);
 	}
 
+	private void ackAsync(String subscriptionName, Collection<String> ackIds) {
+		AcknowledgeRequest acknowledgeRequest = AcknowledgeRequest.newBuilder()
```

Comment:
This is duplicating code in `ack()`. There's only one line difference in all of these new `private` implementation methods. You can reduce code duplication by just changing the `private` methods to have an additional `boolean async` parameter.

Reply:
That's all OK, but see my comment about `ListenableFuture`